modules/miner/blockmanager.go

```diff
@@ -69,6 +70,9 @@ func (m *Miner) newSourceBlock() {
 // it is typically safe to assume that headers will be remembered for
 // min(10 minutes, 10e3 requests).
 func (m *Miner) HeaderForWork() (types.BlockHeader, types.Target, error) {
+	if m.tg.IsStopped() {
+		build.Critical(errClosed)
+	}
```

Comment:
`HeaderForWork` has an error in the function definition, you should be returning `errMinerClosed` here.